src/notebookConverter.ts

```diff
@@ -687,13 +681,24 @@ export class NotebookConverter implements Disposable {
         if (this.notebookFilter.test(doc.uri.fsPath)) {
             const key = NotebookConverter.getDocumentKey(doc.uri);
             const wrapper = this.getTextDocumentWrapper(doc.uri);
-            this.activeDocuments.delete(key);
-            this.activeDocumentsOutgoingMap.delete(NotebookConverter.getDocumentKey(wrapper.uri));
             this.pendingCloseDocuments.set(key, wrapper);
-            wrapper.dispose();
+            this.deleteWrapper(wrapper);
         }
     }
 
+    private deleteWrapper(wrapper: NotebookConcatDocument) {
+        // Cleanup both maps and dispose of the wrapper (disconnects the cell change emitter)
+        this.activeDocumentsOutgoingMap.delete(NotebookConverter.getDocumentKey(wrapper.uri));
+        this.activeDocuments.delete(wrapper.key);
+        wrapper.dispose();
+    }
+
+    private arePathsSame(path1: string, path2: string): boolean {
```

Comment:
This is the only thing from IFileSystem that was used. Didn't really need IFileSystem just to do path comparison